frontend/src/views/instructor/assignments/index.tsx

```diff
@@ -50,7 +90,69 @@ export function InstructorAssignmentsView() {
                     have rejected an offer or had their offer withdrawn will now
                     show up.
                 </p>
-                <InstructorAssignmentsTable />
+                <InstructorAssignmentsTable
+                    position_id={activePosition?.id || -1}
+                    onView={(ddah_id) => {
+                        const ddah = ddahs.find((d) => d.id === ddah_id);
+                        if (ddah) {
+                            setPreviewDdah(ddah);
+                            setPreviewVisible(true);
+                        }
+                    }}
+                    onCreate={(assignment_id) => {
+                        const assignment = assignments.find(
+                            (a) => a.id === assignment_id
+                        );
+                        if (!assignment) {
+                            console.warn(
+                                "Could not find assignment with id",
+                                assignment_id
+                            );
+                            return;
+                        }
+                        const newDdah: Omit<Ddah, "id"> = {
+                            duties: [],
+                            approved_date: null,
+                            accepted_date: null,
+                            revised_date: null,
+                            emailed_date: null,
+                            signature: null,
+                            url_token: "",
+                            total_hours: 0,
+                            assignment,
+                            status: null,
+                        };
+                        setPreviewDdah(newDdah);
+                        setNewDialogVisible(true);
+                    }}
+                    onEmail={() => setEmailDialogVisible(true)}
+                />
+                <DdahPreviewModal
+                    ddah={previewDdah}
+                    show={previewVisible}
+                    onHide={() => setPreviewVisible(false)}
+                    onEdit={async (newDdah: Ddah) => {
+                        await dispatch(upsertDdah(newDdah));
+                    }}
+                />
+                <DdahPreviewModal
```

Comment:
If these modals were global - we could display them from anywhere using Redux store - a pattern I've seen used successfully before if we decide to do so. This would allow us to package this modal display with the view/edit cell component and reduce repetition and complexity